addon/task.go

```diff
@@ -137,32 +137,41 @@ func (h *Task) Completed(n int) {
 }
 
 //
-// deleteReport deletes the task report.
-func (h *Task) deleteReport() {
+// Bucket returns the bucket path.
+func (h *Task) Bucket() (b string) {
+	r := &api.Task{}
 	params := Params{
 		api.ID: h.secret.Hub.Task,
 	}
-	path := params.inject(api.TaskReportRoot)
-	err := h.client.Delete(path)
+	path := params.inject(api.TaskRoot)
+	err := h.client.Get(path, r)
 	if err != nil {
 		panic(err)
 	}
+	b = r.Bucket
+	return
 }
 
 //
-// Bucket returns the bucket path.
-func (h *Task) Bucket() (b string) {
-	r := &api.Task{}
+// Result report addon result.
+func (h *Task) Result(object interface{}) {
```

Comment:
Just the Result is new.
The other diffs are just a result of moving 2 methods.  I like grouping un-exported methods together at the end.